core/storage/feedback/gae_models_test.py

```diff
@@ -49,6 +52,88 @@ def test_put_function(self):
         feedback_thread_model.put()
         self.assertNotEqual(feedback_thread_model.last_updated, last_updated)
 
+    def test_raise_exception_by_mocking_collision(self):
+        feedback_thread_model_cls = feedback_models.GeneralFeedbackThreadModel
+        # Test create method.
+        with self.assertRaisesRegexp(
+            Exception, 'Feedback thread ID conflict on create.'):
+            # Swap dependent method get_by_id to simulate collision every time.
+            with self.swap(
+                feedback_thread_model_cls, 'get_by_id',
+                types.MethodType(
+                    lambda x, y: True,
+                    feedback_thread_model_cls)):
+                feedback_thread_model_cls.create(
+                    'exploration.exp_id.thread_id')
+
+        # Test generate_new_thread_id method.
+        with self.assertRaisesRegexp(
+            Exception,
+            'New thread id generator is producing too many collisions.'):
+            # Swap dependent method get_by_id to simulate collision every time.
+            with self.swap(
+                feedback_thread_model_cls, 'get_by_id',
+                types.MethodType(
+                    lambda x, y: True,
+                    feedback_thread_model_cls)):
+                (
+                    feedback_thread_model_cls
+                    .generate_new_thread_id('exploration', 'entity_id')
+                )
+
+
+class GeneralFeedbackMessageModelTests(test_utils.GenericTestBase):
+    """Tests for the GeneralFeedbackMessageModel class."""
+
+    def test_raise_exception_by_mocking_collision(self):
+        with self.assertRaisesRegexp(
+            Exception, 'Feedback message ID conflict on create.'):
+            # Swap dependent method get_by_id to simulate collision every time.
+            with self.swap(
+                feedback_models.GeneralFeedbackMessageModel, 'get_by_id',
+                types.MethodType(
+                    lambda x, y: True,
+                    feedback_models.GeneralFeedbackMessageModel)):
+                feedback_models.GeneralFeedbackMessageModel.create(
+                    'thread_id', 'message_id')
+
+    def test_get_all_messages(self):
```

Comment:
This test would have more significance if you have more than one message.

Reply:
Done